src/components/LayerSwitcher/AddCustomLayer.tsx

```diff
@@ -259,6 +281,12 @@ const CustomChoose: React.FC<{
         paddingTop: '2rem',
       }}
     >
+      <MissingParts
+        x={url.includes('{x}')}
+        y={url.includes('{y}')}
+        z={url.includes('{z}')}
```

Comment:
Consider placing this component into the `TextField` `helperText` prop. It is usually used for explanations about a field.